entity-service/src/main/kotlin/com/egm/stellio/entity/web/EntityHandler.kt

```diff
@@ -236,6 +244,22 @@ class EntityHandler(
             ResponseEntity.status(HttpStatus.MULTI_STATUS).body(updateResult)
     }
 
+    // TODO add support for multi attribute
+    private fun extractDatasetIdFromNgsiLdAttributes(
+        ngsiLdAttributes: List<NgsiLdAttribute>,
+        attributeName: String
+    ): URI? {
+        val ngsiLdAttribute = ngsiLdAttributes.first {
+            it.name == attributeName
+        }
+        return when (ngsiLdAttribute) {
+            is NgsiLdRelationship -> ngsiLdAttribute.instances[0].datasetId
+            is NgsiLdProperty -> ngsiLdAttribute.instances[0].datasetId
```

Comment:
could be is NgsiLdRelationship, NgsiLdProperty,NgsiLdGeoProperty  -> ngsiLdAttribute.instances[0].datasetId